app/views/course/achievement/course_users/_course_users_form.html.slim

```diff
@@ -2,37 +2,16 @@
   = f.error_notification
   = f.error :achievement
 
-  - collection = course.course_users.without_phantom_users.students.order_alphabetically
-  table.table.table-hover
-    thead
-      th = t('.name')
-      th
-        input.toggle-all id="toggle-all-normal-achievements" type='checkbox' style='display: none'
-        '
-        label for='toggle-all-achievements'
-          = t('.enabled')
-    tbody
-      = f.collection_check_boxes :course_user_ids, collection, :id, :name do |f|
-        tr
-          th = link_to_course_user(f.object)
-          td = f.check_box 'data-for-select-all' => "toggle-all-normal-achievements"
+  - own_students = current_user.my_students(current_course).without_phantom_users
+  - students = course.course_users.without_phantom_users.students
```

Comment:
Can you see if each `course.course_users` call invokes a separate SQL query? 

It might be easier to define a scope by defining it in the controller first, ie. `@course_users = current_course.course_users.order_alphabetically`, then the queries in the view can be simpler, like: 

```ruby
own_students = @course_users.merge(current_course_user.my_students)
students = @course_users.without_phantom_users
phantom_students = @course_users.phantom
```
This will make this file a little more readable!